aten/src/ATen/native/cpu/LerpKernel.cpp

```diff
@@ -0,0 +1,68 @@
+#include <ATen/ATen.h>
+
+#include <ATen/Dispatch.h>
+#include <ATen/native/Lerp.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+namespace at {
+namespace native {
+namespace {
+
+static void lerp_kernel_scalar(
+    Tensor& ret,
+    const Tensor& self,
+    const Tensor& end,
+    Scalar weight) {
+  auto builder = at::TensorIterator::Builder();
+  builder.add_output(ret);
+  builder.add_input(self);
+  builder.add_input(end);
+  auto iter = builder.build();
+
+  AT_DISPATCH_FLOATING_TYPES(ret.scalar_type(), "lerp_kernel_scalar", [&] {
+    scalar_t weight_val = weight.to<scalar_t>();
+    auto weight_vec = Vec256<scalar_t>(weight_val);
+    auto reversed_weight_vec = Vec256<scalar_t>(1 - weight_val);
+    at::native::binary_kernel_vec(
+        *iter,
+        [=](scalar_t self_val, scalar_t end_val) {
+          return (weight_val < 0.5)
+              ? self_val + weight_val * (end_val - self_val)
+              : end_val - (end_val - self_val) * (1 - weight_val);
+        },
+        [=](Vec256<scalar_t> start_vec, Vec256<scalar_t> end_vec) {
+          return reversed_weight_vec * start_vec + weight_vec * end_vec;
```

Comment:
I know this is a WIP. It may seem that non-vectorized implementation above is unnecessarily complicated, however, this implementation is not monotonic on weight, while the implementation above is https://math.stackexchange.com/questions/907327/accurate-floating-point-linear-interpolation/1798323#1798323

Reply:
Right, I will create another PR and drop vec_kernel as not monotonic.